src/main/cms/cms_menu_vtx_smartaudio.c

```diff
@@ -169,18 +173,28 @@ if (saDevice.version == 2) {
     saCmsStatusString[9] = ' ';
 
     if (saDevice.mode & SA_MODE_GET_PITMODE) {
+        saCmsPit = 2;
         saCmsStatusString[10] = 'P';
-        if (saDevice.mode & SA_MODE_GET_IN_RANGE_PITMODE) {
-            saCmsStatusString[11] = 'I';
-        } else {
+        if (saDevice.mode & SA_MODE_GET_OUT_RANGE_PITMODE) {
             saCmsStatusString[11] = 'O';
+        } else {
+            saCmsStatusString[11] = 'I';
         }
         saCmsStatusString[12] = 'R';
         saCmsStatusString[13] = 0;
     } else {
-        int index = (saDevice.version == 2) ? saDevice.power : saDacToPowerIndex(saDevice.power);
-        tfp_sprintf(&saCmsStatusString[10], "%s", vtxCommonLookupPowerName(vtxCommonDevice(), index + 1));
+        saCmsPit = 1;
+        uint8_t powerIndex = 0;
+        bool powerFound = vtxCommonGetPowerIndex(vtxCommonDevice(), &powerIndex);
+        tfp_sprintf(&saCmsStatusString[10], "%s", powerFound ? vtxCommonLookupPowerName(vtxCommonDevice(), powerIndex) : "???");
+    }
+#if defined(USE_VTX_TABLE)
+    if (vtxCommonDevice()->capability.bandCount == 0 || vtxCommonDevice()->capability.powerCount == 0) {
+        strncpy(saCmsStatusString, "PLEASE CONFIGURE VTXTABLE", sizeof(saCmsStatusString));
+        //This string is longer than 13 chars, which means once vtxtable is configured the last few characters may
```

Comment:
They should by all means not linger around. If this is the case then the code does not properly zero terminate the status string when it is set in the next iteration, and this should be fixed.

Reply:
Turns out it doesn't linger because the last part of the status string is the power indication and that gets written either with `tfp_sprintf` or manually with explicit null termination. I will just remove my incorrect comment.

Reply:
Even better. :-)